### DISCUSSION TITLE
Bring data provider functionality to an automation step - DONE

### DISCUSSION THREAD

**Discussion by mjashanks** [Budibase/budibase#2024] [Ideas]
Having recently constructed a few different apps, I ran into the same issue. I need to be able to fetch data (internal table) from an automation action.

I feel like if we were to implement the identical behaviour in an action, as exists in a `Data Provider` it would cover a few crucial needs.

### Enrich relationships

An automation action that could fetch a record where `_id = step1.Row.MyRelationShip` could be used to pull values from related records.

### Fetch all records where condition is met

E.g. on a daily schedule, I want to fetch all invoices that have passed their due-date - then send out an email reminder. I understand that this would also require a way to iterate all results of a collection - i.e. spawning many more automation steps.

### Pull data from other sources too...

1. Views. Currently we cannot use views with automations. 
2. External data sources. We do have an external data sources action, but it does not work well with the new `Datasource Plus` options format.
3. Users. This would enable us to search the user table for certain users - e.g.  "Managers", and assign things / email them accordingly

**Comment by Fabian1410** [comment]
We need this :D

**Reply by mjashanks** [reply]
Coming in the September release!
